src/Microsoft.AspNet.Mvc.Core/MvcRouteHandler.cs

```diff
@@ -40,49 +40,55 @@ public VirtualPathData GetVirtualPath([NotNull] VirtualPathContext context)
             MvcServicesHelper.ThrowIfMvcNotRegistered(services);
 
             EnsureLogger(context.HttpContext);
-            using (_logger.BeginScope("MvcRouteHandler.RouteAsync"))
-            {
-                var actionSelector = services.GetRequiredService<IActionSelector>();
-                var actionDescriptor = await actionSelector.SelectAsync(context);
+            var actionSelector = services.GetRequiredService<IActionSelector>();
+            var actionDescriptor = await actionSelector.SelectAsync(context);
 
-                if (actionDescriptor == null)
+            if (actionDescriptor == null)
+            {
+                string currentRouteValues = string.Empty;
+                if (context.RouteData != null
+                    && context.RouteData.Values != null)
                 {
-                    LogActionSelection(actionSelected: false, actionInvoked: false, handled: context.IsHandled);
-                    return;
+                    currentRouteValues = string.Join(", ",
+                        context.RouteData.Values.Select(rv => $"Key={rv.Key};Value={rv.Value}"));
                 }
 
-                // Replacing the route data allows any code running here to dirty the route values or data-tokens
-                // without affecting something upstream.
-                var oldRouteData = context.RouteData;
-                var newRouteData = new RouteData(oldRouteData);
+                _logger.LogVerbose(
+                    $"No actions matched the current request. Current Route values: {currentRouteValues}");
+                return;
+            }
+
+            _logger.LogVerbose("Action '{ActionName}' matched the request.", actionDescriptor.DisplayName);
```

Comment:
You can move this down inside the scope (see the code sample I sent a while ago via mail).